src/pages/home/components/product-filters/index.tsx

```diff
@@ -0,0 +1,215 @@
+import { Box, Button, Typography } from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useGetFiltersQuery } from "~/redux/filters/filters-api";
+import { setFilters } from "~/redux/filters/filters.slice";
+import { useAppDispatch, useAppSelector } from "~/redux/hooks";
+import { Filters } from "~/libs/types/filters.ts";
+import { ColorPicker } from "../index";
+import { type SelectChangeEvent } from "@mui/material";
+import {
+	CustomSelect,
+	CustomSlider,
+	CustomRadioButtons,
+	CustomRadioButtonGroup,
+} from "~/components";
+import { StyledFiltersContainer } from "./styles";
+import { colors as fontColors, fontSizes } from "~/libs/constants/index.ts";
+
+const maxProductPrice = 10000;
+
+type ProductFiltersProps = {
+	onApply: (filters: Filters) => void;
+	onClear: () => void;
+	initialFilters: Filters;
+};
+
+const ProductFilters: React.FC<ProductFiltersProps> = ({
+	onApply,
+	onClear,
+	initialFilters,
+}) => {
+	const dispatch = useAppDispatch();
+	const filterValues = useAppSelector((state) => state.filters);
+
+	const { t } = useTranslation();
+
+	const { brands, colors, sizes, materials, styles } = filterValues;
+	const { data, error, isLoading } = useGetFiltersQuery({});
+
+	const [minPrice, setMinPrice] = useState<number>(
+		initialFilters.minPrice || 0,
+	);
+	const [maxPrice, setMaxPrice] = useState<number>(
+		initialFilters.maxPrice || 10000,
+	);
+	const [selectedBrand, setSelectedBrand] = useState<string>(
+		initialFilters.brand ? String(initialFilters.brand) : "",
+	);
+	const [selectedColor, setSelectedColor] = useState<string>(
+		initialFilters.color ? String(initialFilters.color) : "",
+	);
+	const [selectedSize, setSelectedSize] = useState<string>(
+		initialFilters.size ? String(initialFilters.size) : "",
+	);
+	const [selectedMaterial, setSelectedMaterial] = useState<string>(
+		initialFilters.material ? String(initialFilters.material) : "",
+	);
+	const [selectedStyle, setSelectedStyle] = useState<string>(
+		initialFilters.style ? String(initialFilters.style) : "",
+	);
+	const [selectedGender, setSelectedGender] = useState<string>(
+		initialFilters.gender ? String(initialFilters.gender) : "",
+	);
+
+	useEffect(() => {
+		if (data) {
+			dispatch(setFilters(data));
+		}
+	}, [data, dispatch]);
+
+	const handlePriceChange = (_: Event, newValue: number | number[]): void => {
+		if (Array.isArray(newValue)) {
+			setMinPrice(newValue[0]);
+			setMaxPrice(newValue[1]);
+		}
+	};
+
+	const handleMinPriceChange = (
+		event: React.ChangeEvent<HTMLInputElement>,
+	): void => {
+		setMinPrice(+event.target.value);
+	};
+
+	const handleMaxPriceChange = (
+		event: React.ChangeEvent<HTMLInputElement>,
+	): void => {
+		setMaxPrice(+event.target.value);
+	};
+
+	const handleApply = (): void => {
```

Comment:
Did you run lint? You have no error "JSX props should not use arrow functionseslintreact/jsx-no-bind"?

Reply:
fixed all linter errors. I didn't know that I should write this command. 
I will try to refactor product filters component later